content/news/XXXX-XX-XX-gsoc-final-blog-fatih-emre-yildiz.md

```diff
@@ -0,0 +1,395 @@
+
+Let's think about another scenario. You love to find out remixes of each track in your library and with those remixes you expend your library. You probably use SoundCloud or a similar service very often. Luckily we have this service in our Find On Web feature. If you are looking for Remixes of tracks in your library very often, Find On Web feature will help you along with that.
+
+This feature is placed in the Track Menu. When you right-click on a track, you will see different Menus and Actions that you can take over a track. This is called `Track Menu`. On the track menu, in the Metadata section, you can see the Find On Web. There are three main Online Music Databases that exist as menus. They will be called services after this part. These services are:
+
+* SoundCloud
+* Discogs
+* Last.fm
+
+When you hover your mouse over those services-menus, you can see the related track properties as actions. Each action populates the suggested Service with the suggested Track Information on a default browser. This looks like:
+
+*Find On Web Feature*
+![Find On Web Feature]({static}/images/news/findonwebfeature.png)
+
+This Implementation was designed roughly for the next step. The next step was:
+
+##### *Find On Web With Factory Design Pattern*
+This implementation aims to have a Factory Design Pattern for the Find On Web feature. 
+
+As its name states, you can think about you have a factory and you can produce products. Our factory is the Find On Web itself and our products are the services. The most important point here is that all products coming out of a factory must be derived from the same interface or class. Thanks to this pattern, you can abstract our Services for future use. you can produce products - add new services - straight forward.
+
+Since there are many other online music databases, that is nice to have Find On Web feature with the Factory Design Pattern. Just in case if needed any other service can be added easily without making any changes in the code base.
+
+Related Files:
+
+* [findonwebfactory.cpp](https://github.com/mixxxdj/mixxx/blob/main/src/widget/findonwebmenufactory.cpp)
+* [findonwebfactory.h](https://github.com/mixxxdj/mixxx/blob/main/src/widget/findonwebmenufactory.h)
+* [Find On Web Menu Services Folder](https://github.com/mixxxdj/mixxx/tree/main/src/widget/findonwebmenuservices)
+
+##### *Editable Track Menu*
+While discussing the `Find On Web` feature on Zulip. There was an idea which was the Track Menu getting crowded and we might consider making it editable. At first I thought the same and this feature would be nice to have and that can help me to get to know the code base for the preference option. So I tried to make it, just to learn how to deal with the preferences. After I introduced this idea on the Zulip channel, that was rejected by the community. So we have decided to not implement this.
```

Comment:
You can write something: "The original idea of making the whole Track Menu customizable has been rejected in discussion with the community in favor of good defaults. Sometimes a configuration option just proves the inability of the developers the decide for them. So we have decided to not implement this."

Reply:
That is way easier to understand and more descriptive at the same time! Thanks.